pkg/controller/source/source.go

```diff
@@ -45,21 +51,129 @@ type Source interface {
 	Start(eventhandler.EventHandler, workqueue.RateLimitingInterface, ...predicate.Predicate) error
 }
 
+var _ Source = &ChannelSource{}
+
 // ChannelSource is used to provide a source of events originating outside the cluster
```

Comment:
I am trying to understand the correct usage pattern for the channel-source so that we can advise users accordingly.

For external triggers which maps to changes in a `type` in K8s, controller will automatically pick them up because it is watching that type.
Now left are the ones which don't result in any data change in the type directly but impact the type, channel source is the right fit for those.

We have to come up with a better example demonstrating this source.